cluster-up/cluster/k3d/common.sh

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env bash
+#
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2022 The Kepler Contributors
+#
+
+set -ex pipefail
+
+CTR_CMD=${CTR_CMD-docker}
+
+CONFIG_PATH="cluster-up/cluster"
+K3D_VERSION=${K3D_VERSION:-5.4.6}
+K3D_MANIFESTS_DIR="$CONFIG_PATH/${CLUSTER_PROVIDER}/manifests"
+CLUSTER_NAME=${K3D_CLUSTER_NAME:-kepler}
+CLUSTER_NETWORK="kepler-network"
+CLUSTER_SERVERS=${K3D_SERVERS:-1}
+CLUSTER_AGENTS=${K3D_AGENTS:-1}
+LOADBALANCER_PORT=${LOADBALANCER_PORT:-8081}
+
+REGISTRY_NAME=${REGISTRY_NAME:-kepler-registry-local}
+REGISTRY_PORT=${REGISTRY_PORT:-5001}
+
+IMAGE_REPO=${IMAGE_REPO:-localhost:5001/kepler}
+IMAGE_TAG=${IMAGE_TAG:-devel}
+
+PROMETHEUS_OPERATOR_VERSION=${PROMETHEUS_OPERATOR_VERSION:-v0.11.0}
+PROMETHEUS_ENABLE=${PROMETHEUS_ENABLE:-true}
+PROMETHEUS_REPLICAS=${PROMETHEUS_REPLICAS:-1}
+GRAFANA_ENABLE=${GRAFANA_ENABLE:-true}
+
+CONFIG_OUT_DIR=${CONFIG_OUT_DIR:-"_output/manifests/${CLUSTER_PROVIDER}/generated"}
+rm -rf ${CONFIG_OUT_DIR}
+mkdir -p ${CONFIG_OUT_DIR}
+
+# check CPU arch
+PLATFORM=$(uname -m)
+case ${PLATFORM} in
+x86_64* | i?86_64* | amd64*)
+    ARCH="amd64"
+    ;;
+ppc64le)
+    ARCH="ppc64le"
+    ;;
+aarch64* | arm64*)
+    ARCH="arm64"
+    ;;
+*)
+    echo "invalid Arch, only support x86_64, ppc64le, aarch64"
+    exit 1
+    ;;
+esac
+
+function _get_prometheus_operator_images {
+    grep -R "image:" kube-prometheus/manifests/*prometheus-* | awk '{print $3}'
+    grep -R "image:" kube-prometheus/manifests/*prometheusOperator* | awk '{print $3}'
+    grep -R "prometheus-config-reloader=" kube-prometheus/manifests/ | sed 's/.*=//g'
+    if [ ${GRAFANA_ENABLE,,} == "true" ]; then
+        grep -R "image:" kube-prometheus/manifests/*grafana* | awk '{print $3}'
+    fi
+}
+
+function _load_prometheus_operator_images_to_local_registry {
+    for img in $(_get_prometheus_operator_images); do
+        $CTR_CMD pull $img
+        $CTR_CMD tag $img "localhost:${REGISTRY_PORT}/${img}"
+        $CTR_CMD push "localhost:${REGISTRY_PORT}/${img}"
+    done
+} 
+
+function _deploy_prometheus_operator {
```

Comment:
... en I suppose for any k8s, not matter kind or k3s, we should use same steps to deploy prometheus right?

Reply:
@SamYuan1990 i have made no change in the prometheus and grafana provisioning related functions, there are identical with the ones from kind